app/controllers/reports_controller.rb

```diff
@@ -20,6 +20,7 @@ def total_inventory_value
   def graphs
     @report_by_date = Reports::Graphs.order_count_by_day
     @report_by_month = Reports::Graphs.order_count_by_month
+    @report_items_by_month = Reports::Graphs.item_count_by_month
```

Comment:
All these graphs are being loaded even when they aren't used. I think you should change `Graphs` into an instance object, so you can just have a single `@graphs = Reports::Graphs.new` method. Then each of the current `self.` methods could just be instance methods. If the methods are possibly called more than once, then make sure to lazy load them like so:
```
def item_count_by_month
  @item_count_by_month ||=
    begin
      # Load the data with the last line returning the actual data
    end
end
```
If they are only ever called once in the view, then you can leave them as not lazy loaded.

Reply:
Firstly, I completely agree.

However, I'd like to make this part of an improvement for all graphs. This is more involved of a refactor than I was hoping to do at this moment if thats ok.